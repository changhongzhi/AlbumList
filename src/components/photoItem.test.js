import React from 'react';
import renderer from 'react-test-renderer';
import PhotoItem from './photoItem';

test('render default not show image', () => {
    const viewport = {top: 500, height: 950};
  const component = renderer.create(
    <PhotoItem title="photo title" url="url" thumbnailUrl="thumbnailUrl" viewport={viewport}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('render when show image', () => {
    const viewport = {top: 500, height: 950};
  const component = renderer.create(
    <PhotoItem title="photo title" url="url" thumbnailUrl="thumbnailUrl" showImage={true} viewport={viewport}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should be able to update position ', () => {
    const viewport = {top: 500, height: 950};
  const component = renderer.create(
    <PhotoItem title="photo title" url="url" thumbnailUrl="thumbnailUrl" viewport={viewport} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
