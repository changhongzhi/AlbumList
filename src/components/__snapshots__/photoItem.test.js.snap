exports[`test render default not show image 1`] = `
<div>
  <h2>
    photo title
  </h2>
  <img
    alt="photo title"
    src="thumbnailUrl" />
</div>
`;

exports[`test render when show image 1`] = `
<div>
  <h2>
    photo title
  </h2>
  <img
    alt="photo title"
    src="url" />
</div>
`;

exports[`test should be able to update position  1`] = `
<div>
  <h2>
    photo title
  </h2>
  <img
    alt="photo title"
    src="thumbnailUrl" />
</div>
`;
